### PR TITLE
Variables: Add confirmation modal when deleting variables

### PR DESCRIPTION
**Pull request by kaydelaney**
**What this PR does / why we need it**:
Adds a confirmation modal when a user tries to delete a template variable.

**Which issue(s) this PR fixes**:
 See #52832

### CODE REVIEW

File: public/app/features/variables/editor/ConfirmDeleteModal.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  varName: string;
+  isOpen: boolean;
+  onConfirm: () => void;
+  onDismiss: () => void;
+}
+
+export function ConfirmDeleteModal({ varName, isOpen = false, onConfirm, onDismiss }: Props) {
+  return (
+    <ConfirmModal
+      title="Delete"
```
Comment (axelavargas) [inline public/app/features/variables/editor/ConfirmDeleteModal.tsx:16]:
```suggestion
      title="Delete variable"
```

Nit: more clarity to the users in where they are

**Review by axelavargas — COMMENTED** [review]
Awesome job, @kaydelaney 👏🏾 I left a small nit.  Also, while checking your PR I found a small bug that maybe you can tackle in your PR (we can open a new PR if this is too much work)

Reproduce steps:
1. Delete a variable
2. Try to create a new variable, the editor is not opening, but the variable is created

Bug

**Review by axelavargas — APPROVED** [review]
🥳🥳 🥳  great work, @kaydelaney

**Comment by torkelo** [comment]
@kaydelaney did you consider using the ConfirmButton? It's what we use for delete actions in other lists views and feels better than a big modal https://developers.grafana.com/ui/canary/index.html?path=/story/buttons-confirmbutton--delete

**Pull request merged**